mesonbuild/environment.py

```diff
@@ -272,6 +272,9 @@ def __init__(self, source_dir, build_dir, main_script_launcher, options, origina
         if mesonlib.is_windows():
             self.default_c = ['cl', 'cc', 'gcc', 'clang']
             self.default_cpp = ['cl', 'c++', 'g++', 'clang++']
+        elif mesonlib.is_freebsd():
+            self.default_c = ['clang', 'gcc', 'cc']
+            self.default_cpp = ['clang++', 'g++', 'c++']
```

Comment:
Is this really needed? cc should be a hard link to clang.

Reply:
Since I've upgraded cc is clang, but initially it was gcc 4.2.1. At this point we can probably drop this.

Reply:
FreeBSD 10 and later versions use clang by default and the oldest supported release is 10, so I think gcc is only used by tier-2 and tier-3 platforms.